File: 1009-complement-of-base-10-integer/1009-complement-of-base-10-integer.go
func bitwiseComplement(n int) int {
    if n == 0 {
        return 1
    }
    t := n
    o := 0
    for t > 0 {
        t >>= 1
        o = (o << 1) + 1
    }
    return o ^ n
}

File: 1009-complement-of-base-10-integer/1009-complement-of-base-10-integer.java
class Solution {
    public int bitwiseComplement(int n) {
        if (n == 0) {
            return 1;
        }
        int temp = n;
        int other = 0;
        while (temp > 0) {
            temp >>= 1;
            other = (other << 1) + 1;
        }
        return other ^ n;
    }
}